insights/mappers/certificates_enddate.py

```diff
@@ -67,3 +72,18 @@ def get_expiration_date(self, path):
                     if self.data and path in self.data else None)
         except:
             raise Exception("Unable to parse the expiration data of %s" % path)
+
+    def expiration_date(self, path):
+        """
+        :param path: the certificate file path
+        :returns: return a namedtuple(['str', 'datetime']) contains the
+                  expiration date in string and datetime format.
+        :raises Exception: unparsable expiration data
+        """
+        path_date = self.data.get(path)
+        if path_date:
+            try:
+                path_datetime = datetime.strptime(path_date, '%b %d %H:%M:%S %Y')
+                return self.ExpirationDate(path_date, path_datetime)
+            except:
+                raise Exception("Unparsable expiration data of %s" % path)
```

Comment:
> If it is useful to know that the datestamp is invalid then the better approach would be to keep only valid data in this mapper, and create a separate mapper for invalid datestamp data.    

@bfahr  Hi, I am sorry, I don't understand the meaning of `a separate mapper for invalid datestamp data`, why & how ? :question: